hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElementType.java

```diff
@@ -682,13 +683,25 @@ public Type toType(String propertyName) throws QueryException {
 		}
 	}
 
-	public String getIdentifierPropertyName() {
-		if ( getEntityPersister() != null && getEntityPersister().getEntityMetamodel() != null
-				&& getEntityPersister().getEntityMetamodel().hasNonIdentifierPropertyNamedId() ) {
-			return getEntityPersister().getIdentifierPropertyName();
-		}
-		else {
-			return EntityPersister.ENTITY_ID;
+	public String[] getIdentifierPropertyName() {
```

Comment:
@dreab8 I think you forgot to update this method name: it should be `getIdentifierPropertyNames()` for the sake of consistency.

Thanks!

Reply:
@gsmet 
you are right my mistake.
I'm fixing it
Thanks!